src/Orchard.Web/Modules/Orchard.Projections/Services/ProjectionManager.cs

```diff
@@ -112,9 +116,15 @@ public class ProjectionManager : IProjectionManager{
             }
 
             // aggregate the result for each group query
+            var checkPermissions = queryRecord.FilterGroups.Any(x => x.Filters.Any(y => y.CheckPermissions));
 
-            return GetContentQueries(queryRecord, Enumerable.Empty<SortCriterionRecord>())
-                .Sum(contentQuery => contentQuery.Count());
+            var count = GetContentQueries(queryRecord, Enumerable.Empty<SortCriterionRecord>())
+                .Sum(contentQuery => checkPermissions ? 
+                                contentQuery.List().FilterContentItems(_services.WorkContext.CurrentUser).Count() : 

```

Comment:
What about pages > 1 when the content item which is filtered is on page 1. If it was supported then it would mean checking for ALL items up to the current page. Is that your concern?